scripts/sparse-and-shallow-git-fetch.sh

```diff
@@ -12,8 +12,6 @@ fi
 
 set -ev
 
-GIT="/usr/bin/git"
```

Comment:
I think there's two reasons: stability and security.

Having a full path to all tools you depend on helps with stability because you have fewer dependencies on the outside world. I've seen issues where cron produces a different environment as a result of not loading login scripts. You find out about those when your job runs in cron (ideally) or when there's a production emergency (less than ideal).

The security argument is a bit of a stretch. If an attacker were able to write a file locally (/tmp) and then somehow cause PATH to be updated, they could trigger execution of arbitrary commands as the user running the server. Forcing /usr/bin/git makes this more difficult, because /usr/bin is typically not writable by anyone but root (and your service had better not be running as root...).

rm and mkdir below should probably also have full paths.